NMS-14189: Document various new UI features (#4607)
OpenNMS/opennms: docs/modules/reference/pages/configuration/file-editor.adoc

[[file-editor]]

> = File Editor: Manage Configuration Files from the UI

Author: Changed to "Manage Configuration Files in the UI"

> The File Editor, available in the new UI, allow users to view and edit configuration files (`.xml`, `.cfg`, `.drl`, etc.) with the OpenNMS UI instead of having to edit them directly in the file system (for example in the `/etc` directory).

Author: Fixed

NOTE: Making changes to configuration files will affect how OpenNMS operates and incorrectly editing a file could result in features no longer working or not working as intended.

To access the File Editor, go to the new UI, for example by clicking on "UI Preview", then navigating to "File Editor".

> == Viewing Files

Reviewer: ```suggestion
== View Files
```

Author: Fixed

> After navigating to the File Editor, a tree view of the OpenNMS `/etc` directory is displayed.

Reviewer: ```suggestion
After navigating to the File Editor, the UI displays a tree view of the OpenNMS `/etc` directory.
```

Author: Fixed

From here you can:

- navigate to any file
- expand and collapse subtrees
- search for any part of a file or directory name
- view and edit files

.File Editor main screen
image::configuration/file-editor-main.png[]

> ## Editing Files

Reviewer: ```suggestion
## Edit Files
```

Author: Fixed

> To edit a file, simply start typing within the file text display.

Reviewer: ```suggestion
To edit a file, type in the file text display.
```

Author: Fixed

To save the file, click "Save".

> CAUTION: This will overwrite the file on disk, there is no versioning or backup file support at this time.

Reviewer: ```suggestion
CAUTION: This will overwrite the file on disk; there is no versioning or backup file support at this time.
```

Author: Fixed

> Use the "Reset" button to reload the file from disk into the editor - this will be the text as it was last saved to disk, not necessarily the original text before you began any editing.

Reviewer: ```suggestion
Click the *Reset* button to reload the file from disk into the editor. 
This will be the text as it was last saved to disk, not necessarily the original text before you began editing.
```

Author: Fixed

## Add Files

To add a new file, navigate to a folder and click the plus `+` sign.

> Enter a file name and add some text, then click "Save" to save the file.

Reviewer: ```suggestion
Enter a file name, add some text, and click *Save*.
```

Author: Fixed

> Note, validation will run when you attempt to save the file and may prevent it from being saved. A message will appear at the bottom of the text editor with more information on why validation failed.

Reviewer: ```suggestion
Note that validation will run when you attempt to save the file and may prevent it from being saved. 
A message at the bottom of the text editor provides more information on why validation failed.
```

Author: Fixed

> To delete a file, click on the minus sign `-` to the right of the file name. A confirmation dialog will be displayed, click "Confirm" to delete the file or "Cancel" to cancel the action.

Author: Fixed

> CAUTION: Deleting a file is permanent, the file will be removed from disk and is unrecoverable.

Author: Fixed

> .Adding a new file

Author: Fixed

> image::configuration/file-editor-add-new.png[]

Author: Fixed to 300

> ## Activating the Changes

Author: Fixed

> To activate any changes made in the File Editor, you may need to restart OpenNMS.

Author: No prompt appears. Could be a UI feature enhancement request.

It depends on which configuration file(s) the user edited and what component(s) it is connected to. Many components do detect config changes and will reload without needing a restart, but some do require a restart. There's no good way to really know this without going through every single file.

I updated this with (maybe?) clearer wording and also a couple of links to topics that provide more info on what features are auto-reloaded. Let me know if that seems ok.

We may want to have a page or update one of those topics to more clearly state which features are auto-reloaded and which not.

> To activate any changes made in the File Editor, you may need to restart OpenNMS.

Reviewer: Sounds good. We do have tables in the reference section that describe daemons and such that need to be restarted and those that don't. We don't want to bog things down here, though. I was just curious. I'll see what you wrote.

## Console

> After performing actions such as "Save" or "Delete", a Console window will appear at the bottom of the edit pane with some detailed information about actions.

Author: Fixed

> You can click "Clear" to clear the Console text or "Minimize" to minimize the Console window.

Author: Fixed
